### PR TITLE
 [CodeCamp] Add module_combination doc

### PR DESCRIPTION
**Pull request by wang-tf**
* fix init_weights not match num_base_priors

 * add module_combination about yolov5 using other model loss

Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation for this PR and the goal you want to achieve through this PR.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit tests to ensure the correctness.
3. If the modification has a potential influence on downstream projects, this PR should be tested with downstream projects, like MMDetection or MMClassification.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: docs/zh_cn/advanced_guides/module_combination.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
+## YOLOV5使用其他的loss
+
+如果想在YOLOV5中使用其他model中的loss，同时保持YOLOV5的网络结构，可以在YOLOV5的配置文件中修改。需要修改配置文件中model的bbox_head部分，将type修改未YOLOv7Head或YOLOXHead，而haed_module中的type则继续使用YOLOv5HeadModule。其中，因为YOLOX是Anchor Free结构，所以需要将head_module中的num_base_priors设置为，同时增加train_cfg中assigner的设置。
```
Comment (PeterH0323) [inline docs/zh_cn/advanced_guides/module_combination.md:3]:
```suggestion
如果想在 YOLOv5 中使用其他 `model` 中的 `loss`，同时保持 YOLOv5 的网络结构，可以在 YOLOv5 的配置文件中修改。需要修改配置文件中 `model` 的 `bbox_head` 部分，将 
 `type` 修改为 `YOLOv7Head` 或 `YOLOXHead`，而 `haed_module` 中的 `type` 则继续使用 `YOLOv5HeadModule`。其中，因为 YOLOX 是 Anchor Free 结构，所以需要将 `head_module` 中的 `num_base_priors` 设置为，同时增加 `train_cfg` 中 `assigner` 的设置。
```

Comment (PeterH0323) [reply docs/zh_cn/advanced_guides/module_combination.md:3]: 有以下几点建议：
1. `num_base_priors` 设置为 后面漏了参数
2. YOLOv5，中间的 v 是小写
3. 英文和中文要有空格隔开
4. 变量名要用 `` 包裹，例如 num_base_priors ->`num_base_priors`

File: docs/zh_cn/advanced_guides/module_combination.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
+## YOLOv5使用其他的loss
```
Comment (hhaAndroid) [inline docs/zh_cn/advanced_guides/module_combination.md:1]: ## model 和 loss 组合替换

在 MMYOLO 中，model 即网络本身和 loss 是解耦的，用户可以简单的通过修改配置文件中 model 和 loss 来组合不同模块。下面给出两个具体例子。

(1) YOLOv5 model 组合 YOLOv7 loss
...

(2) YOLOX model 组合 RTMDet loss
...

File: docs/zh_cn/advanced_guides/module_combination.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
+## model 和 loss 组合替换
+
+在 MMYOLO 中，model 即网络本身和 loss 是解耦的，用户可以简单的通过修改配置文件中 model 和 loss 来组合不同模块。下面给出两个具体例子。
+
+(1) YOLOv5 model 组合 YOLOv7 loss
+
+```python
```
Comment (hhaAndroid) [inline docs/zh_cn/advanced_guides/module_combination.md:7]: 不能写的这么省略，要提供直接可用的配置，因此 _base_ 要写出来

File: docs/zh_cn/advanced_guides/module_combination.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+            num_classes=num_classes,
+            in_channels=[256, 512, 1024],
+            widen_factor=widen_factor,
+            featmap_strides=strides,
+            num_base_priors=3),
+        )
+)
+```
+
+(2) YOLOv5 model 组合 YOLOX loss
```
Comment (hhaAndroid) [inline docs/zh_cn/advanced_guides/module_combination.md:22]: YOLOX model 组合 RTMDet loss

**Review by hhaAndroid — APPROVED** [review]

**Pull request merged**